### ISSUE TITLE
Create Action Plan From Template apex action in flow creates tasks without dependencies

### ISSUE THREAD

**Issue by irinaIvanova97** [SalesforceLabs/ActionPlansV4#151]
**Describe the bug**
When creating a new action plan from template with the apex action "Create Action Plan From Template", the tasks related to the record are created all at once even though there is a dependency defined in the template.

**To Reproduce**
1. Go go Action Plan Templates
2. Create new template
3. Populate name
4. Create new task with sample subjects and days after "1", priority normal, send email true, reminder true
5. Create a second task with sample subjects and days after "1", priority normal, send email true, reminder true. On the controlling task select the task created in the previous step.
6. Save
7. Create new record-triggered flow on lead creation.
8. Call the apex action "Create Action Plan From Template" with the current record id and template name passed
9. Activate flow
10. Create lead record
11. Lead  has 2 task created and assigned to the record

**Expected behavior**
The record to have only the first task, since the second one is dependent on the first and should be generated only after the first one is completed.

**Screenshots**
![image](https://github.com/SalesforceLabs/ActionPlansV4/assets/44892329/a6ac4a73-e7be-477c-b5df-f76d9fca8606)
![image](https://github.com/SalesforceLabs/ActionPlansV4/assets/44892329/305bd2e7-accf-40a9-90f0-f2d7dc416173)

**Environment (please complete the following information):**
 - OS: Windows
 - Browser: Chrome
 - Action Plans Version: 4.3
 - Salesforce UI: Lightning Experience
 - Org Edition: Enterprise

**Comment by charliel1981** [comment]
Hello, 

i believe i got to the bottom of this issue with workbench. You might find that you originally created an AP and deleted rows of tasks out (potentially in the classic UI - as i did) and the dependencies didnt adjust as they should have. 
If you use workbench and look at the action plan template using LabsActionPlans__Action_Plan_Template__c as the filter to the template, more than likely you will see that the LabsActionPlans__Dependent__c may not be correct! 

You can then use workbench to fix this issue by updating the record.

This worked for me, hope it helps anyone with the same issue
